client/src/app/gateways/repositories/polls/vote-repository.service/vote-repository.service.ts

```diff
@@ -42,9 +54,57 @@ export class VoteRepositoryService extends BaseMeetingRelatedRepository<ViewVote
     }
 
     public async hasVotedFor(...ids: Id[]): Promise<HasVotedResponse | undefined> {
-        if (ids.length) {
-            return await this.http.get(`${HAS_VOTED_URL}?ids=${ids.join()}`);
+        if (!ids.length) {
+            return undefined;
+        }
+
+        const result = this.getIdsFromCache(ids);
+        if (ids.some(id => !!result[id])) {
```

Comment:
```suggestion
        if (ids.every(id => !!result[id])) {
```
or else you would return potentially incomplete results